### ISSUE TITLE
Supporting Promise API

### ISSUE THREAD

**Issue by screendriver** [jimhigson/oboe.js#75]
Are there any plans to support the new [Promise API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)?

**Comment by JuanCaicedo** [comment]
I'm not entirely sure how that would work, since Oboe registers a bunch of listeners instead of a callback for when a single operation completes. Do you have an idea of where you would expect to be able to use them?

**Comment by JuanCaicedo** [comment]
I would think @jimhigson probably wants to keep Oboe minimal and not dependent on having ES6 in the browser, but it might be possible to work those functions so that they're easy to promisify. Maybe all it would take is modifying the object returned by `oboe()` so that it can be passed into something like `Bluebird.promisify()`

**Comment by yckart** [comment]
Yeah, I really like this idea! It was my first thought, as I saw the oboe API first. Would nice to have this implemented, some day...

**Comment by Aigeec** [comment]
Could we allow someone to provide a promise library. 
Then we could use the provided one, if none has been provided then check for native and finally error with "Unsupported Operation" if some one is using "then" but we can't find a promise?
